GitExtUtils/GitUI/TaskManager.cs

```diff
@@ -67,5 +61,36 @@ public void JoinPendingOperations()
             // since the latter adds new tasks to the collection and would therefore never complete.
             JoinableTaskContext.Factory.Run(_joinableTaskCollection.JoinTillEmptyAsync);
         }
+
+        public async Task ReportExceptionOnMainThreadAsync(Exception ex)
```

Comment:
Why public?
Same for HandleExceptionsAsync().
Only used in RunAsyncAndForget(), kept here to show the analogy?

Reply:
`public` because of generic applicability. They might be used in the follow-up PR (announced in the PR description).

Reply:
Can we please make those internal and open up only when such necessity presents itself?